bundles/mapping/mapwfs2/service/WFSLayerService.js

```diff
@@ -139,35 +139,23 @@ Oskari.clazz.define(
          * Handles status of selected features
          */
         setWFSFeaturesSelections: function (layerId, featureIds, makeNewSelection) {
-            var me = this;
-
+            let updatedIds;
             if (makeNewSelection) {
-                this.__removeFeatureSelectionForLayer(layerId);
-                me.getWFSSelections().push({ 'layerId': layerId, 'featureIds': featureIds });
+                updatedIds = featureIds;
             } else {
```

Comment:
Could we remove the if statement and just assign featureIds as updatedIds and reverse the else as a new if?
```
let updatedIds = featureIds;
if (!makeNewSelection) {
```